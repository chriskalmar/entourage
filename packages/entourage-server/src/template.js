import { renderFile } from './render';
import fs from 'fs';
import path from 'path';
import { getWorkVersionFolder } from './util';

// eslint-disable-next-line quotes
const escapePatterns = [`__PORTS\\..+`];

export const renderTemplate = (templateFilename, templateParams) => {
  const filename = templateFilename.startsWith('/')
    ? templateFilename
    : `${path.basename(process.env.PROFILES_PATH)}/${templateFilename}`;

  if (!fs.existsSync(filename)) {
    throw new Error(`Template '${templateFilename}' not found`);
  }

  return renderFile(filename, templateParams, escapePatterns);
};

export const renderTemplateToFile = (
  templateFilename,
  templateParams,
  version,
  outputFilename,
) => {
  const renderedTemplate = renderTemplate(templateFilename, templateParams);

  const filename = `${getWorkVersionFolder(version)}/${outputFilename}`;

  if (!fs.existsSync(path.dirname(filename))) {
    fs.mkdirSync(path.dirname(filename), { recursive: true });
  }

  fs.writeFileSync(filename, renderedTemplate, 'utf8');
};
