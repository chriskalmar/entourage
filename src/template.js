import { renderFile } from './render';
import fs from 'fs';
import path from 'path';

export const renderTemplate = (templateFilename, templateParams) => {
  const filename = templateFilename.startsWith('/')
    ? templateFilename
    : `${path.basename(process.env.PROFILES_PATH)}/${templateFilename}`;

  if (!fs.existsSync(filename)) {
    throw new Error(`Template '${templateFilename}' not found`);
  }

  return renderFile(filename, templateParams);
};

export const renderTemplateToFile = (
  templateFilename,
  templateParams,
  version,
  outputFilename,
) => {
  const renderedTemplate = renderTemplate(templateFilename, templateParams);

  const filename = `${path.basename(
    process.env.WORK_PATH,
  )}/${version}/${outputFilename}`;

  if (!fs.existsSync(path.dirname(filename))) {
    fs.mkdirSync(path.dirname(filename), { recursive: true });
  }

  fs.writeFileSync(filename, renderedTemplate, 'utf8');
};
