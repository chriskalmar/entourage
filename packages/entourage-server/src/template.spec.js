import { renderTemplate } from './template';

describe('tempalte', () => {
  it('should render from a template', () => {
    const result = renderTemplate(`${__dirname}/../test/demo-profile.yaml`, {
      __PROFILE: 'blue-space-omlette',
    });

    expect(result).toMatchSnapshot();
  });
});
