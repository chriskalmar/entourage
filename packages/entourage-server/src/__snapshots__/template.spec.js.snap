// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tempalte should render from a template 1`] = `
"defaults:
  params:
    BASE_DB: \\"base_project_12\\"
    EXTERNAL_ADDRESS: \\"127.0.0.1\\"
    RABBITMQ: \\"3.7.17-management\\"

renderTemplates:
  sourcePath: templates
  targetPath: .
  files:
    - blue-space-omlette.init-db.sh
    - blue-space-omlette.docker-compose.yaml
    - blue-space-omlette.env

prepare:
  timeout: 90
  script:
    - . blue-space-omlette.init-db.sh

docker:
  composeFile: blue-space-omlette.docker-compose.yaml
"
`;
